external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -139,7 +142,16 @@ object SchemaConverters {
 
       case FloatType => builder.floatType()
       case DoubleType => builder.doubleType()
-      case _: DecimalType | StringType => builder.stringType()
+      case StringType => builder.stringType()
+      case d: DecimalType =>
+        val avroType = LogicalTypes.decimal(d.precision, d.scale)
+        if (nullable) {
+          val schema = avroType.addToSchema(SchemaBuilder.builder().bytesType())
```

Comment:
do we loss nullable information here?

Reply:
no, in the next line `builder` is nullable.